ztp/ztp-ran-policy-generator/README.md

```diff
@@ -1,11 +1,265 @@
+
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+ name: $name
+ labels:
+   machineconfiguration.openshift.io/role: $name
+spec:
+ machineConfigSelector:
+   matchExpressions:
+     - key: machineconfiguration.openshift.io/role
+       operator: In
+       values: [worker, $name]
+ nodeSelector:
+   matchLabels:
+     node-role.kubernetes.io/$name: ""
+```
+
+The generated policy will need the $name as input parameter in order to adjust the policy for the specific group requirements (ex:groupN $name = worker-du). The generated policy that will apply groupN MachineConfigPool will be as below
+
+```
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-mcp-du
+  namespace: profile-ran-subscriptions
+  annotations:
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/categories: CM Configuration Management
+    policy.open-cluster-management.io/controls: CM-2 Baseline Configuration
+spec:
+  remediationAction: enforce
+  disabled: false
+  policy-templates:
+    - objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: policy-workerdu-mcp
+        spec:
+          remediationAction: enforce # the policy-template spec.remediationAction is overridden by the preceding parameter value for spec.remediationAction.
+          severity: low
+          namespaceSelector:
+            exclude: ["kube-*"]
+            include: ["*"]
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                apiVersion: machineconfiguration.openshift.io/v1
+                kind: MachineConfigPool
+                metadata:
+                  name: worker-du
+                  labels:
+                    machineconfiguration.openshift.io/role: worker-du
+                spec:
+                  machineConfigSelector:
+                    matchExpressions:
+                      - key: machineconfiguration.openshift.io/role
+                        operator: In
+                        values: [worker, worker-du]
+                  nodeSelector:
+                    matchLabels:
+                      node-role.kubernetes.io/worker-du: ""
+```
+
+3 - RanGenTemplate:
+
+RanGenTemplate is a Custom Resource Definition (CRD) that tells RanGen where to locate the generated policies and which Spec items need to be defined. Check the [RanGenTemplate](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ztp-ran-policy-generator/ranGenTemplates/RanGenTemplate.yaml) for more info. Let's consider the [group-du-ranGen.yaml](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ztp-ran-policy-generator/ranGenTemplateExamples/group-du-ranGen.yaml) example to explain more
+
+```
+apiVersion: ranPolicyGenerator/v1
+kind: RanGenTemplate
```

Comment:
I would suggest some name changing. The tool generates policies from the source objects according to the specification below. It is not specific for RAN. Can it be called PolicyGeneratorSpec?

Reply:
done, rename to PolicyGenerator